Plot.Tests/Components/PartialComponents/Button.cs

```diff
@@ -1,29 +1,93 @@
+using Microsoft.Playwright;
 using Microsoft.Playwright.Xunit;
 using Xunit;
 
-namespace PlaywrightTests;
-
-public class ButtonTest : PageTest
+namespace PlaywrightTests
 {
-    [Fact]
-    public async Task HasButtonAndContent()
+    public class ButtonTests : PageTest
     {
-        // Go to test page.
-        await Page.GotoAsync("http://localhost:8080/test/button");
+        // Test to check that all buttons exist and display the correct content.
+        [Fact]
+        public async Task HasButtonAndContent()
+        {
+            //navigate to page
+            await Page.GotoAsync("http://localhost:8080/test/button");
 
-        // Expect the page to have a body visibile.
-        await Expect(Page.Locator("body")).ToBeVisibleAsync();
+            //locate each button
+            var button1 = Page.Locator("button:has-text('Test Text 1')");
+            var button2 = Page.Locator("button:has-text('Test Text 2')");
+            var button3 = Page.Locator("button:has-text('Test Text 3')");
```

Comment:
@pmdurfee I would not recommend using this as a way to target your elements. Please give each element on the test page a unique ID and target them like that.